solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,668 @@
+  }
+}
+
+variable "key_type" {
+  type        = string
+  description = "The type of private key which is to be generated."
+  default     = "rsa"
+  validation {
+    condition     = contains(["rsa", "ec"], var.key_type)
+    error_message = "The specified key_type is not valid. Allowed values are: rsa, ec. [Learn More](https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-secrets-manager-cli#secrets-manager-configurations-cli)"
+  }
+}
+
+variable "permitted_dns_domains" {
+  type        = list(string)
+  description = "The Allowed DNS domains or subdomains for the certificates which is to be signed and issued by the CA certificate."
+  default     = null
+
+  validation {
+    condition     = var.permitted_dns_domains == null ? true : length(var.permitted_dns_domains) <= 10
+    error_message = "length of permitted_dns_domains must be <= 100"
+  }
+
+  validation {
+    condition = var.permitted_dns_domains == null ? true : alltrue([
+      for permitted_dns_domain in var.permitted_dns_domains : can(regex("^(.*?)$", permitted_dns_domain))
+    ])
+    error_message = "list items must match regular expression /(.*?)/"
+  }
+}
+
+variable "alt_names" {
+  type        = list(string)
+  description = "The alternate names of the certificate which is to be created."
+  default     = null
+
+  validation {
+    condition     = var.alt_names == null ? true : length(var.alt_names) < 100
+    error_message = "length of alt_names must be < 100"
+  }
+
+  validation {
+    condition = var.alt_names == null ? true : alltrue([
+      for alt_name in var.alt_names : can(regex("^(.*?)$", alt_name))
+    ])
+    error_message = "list items must match regular expression /^(.*?)$/"
+  }
+}
+
+variable "ttl" {
+  type        = string
+  description = "The Time-to-live (TTL) which is to be assigned to a private certificate. [Learn More](https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-secrets-manager-cli#secrets-manager-configurations-cli)"
+  default     = null
+  validation {
+    condition     = var.ttl == null ? true : can(regex("^[0-9]+[s,m,h,d]{0,1}$", var.ttl))
+    error_message = "ttl must match regular expression /^[0-9]+[s,m,h,d]{0,1}$/"
+  }
+}
+
+variable "exclude_cn_from_sans" {
+  type        = bool
+  description = "Whether the common name is excluded from Subject Alternative Names (SANs). If set to true, the common name is not included in DNS or Email SANs if they apply"
+  default     = false
+}
+
+##############################################################################
+# ROOT CA
+##############################################################################
+
+variable "root_ca_name" {
+  type        = string
+  description = "The name of the Root CA to be created for a private_cert secret engine."
```

Comment:
the description needs to be updated to mention about the naming convention used when using prefix